src/eduid_webapp/reset_password/views/reset_password.py

```diff
@@ -0,0 +1,369 @@
+    * Create a PasswordResetEmailState in the password_reset_state_db
+      (holding the email address, the eppn of the user associated to the
+      email address in the central userdb, and a freshly generated random hash
+      as an identifier code for the created state);
+    * Email the generated code to the received email address.
+    
+    The operation can fail due to:
+    * The email address does not correspond to any valid user in the central db;
+    * There is some problem sending the email.
+    """
+    current_app.logger.info(f'Trying to send password reset email to {email}')
+    try:
+        send_password_reset_mail(email)
+    except MailTaskFailed as error:
+        current_app.logger.error(f'Sending password reset e-mail for {email} failed: {error}')
+        return error_message('resetpw.send-pw-fail')
+
+    return success_message('resetpw.send-pw-success')
+
+
+@reset_password_views.route('/config/', methods=['POST'])
+@UnmarshalWith(ResetPasswordEmailCodeSchema)
+@MarshalWith(FluxStandardAction)
+def config_reset_pw(code: str) -> dict:
+    """
+    View that receives an emailed reset password code and returns the
+    configuration needed for the reset password form.
+
+    Preconditions required for the call to succeed:
+    * A PasswordResetEmailState object in the password_reset_state_db
+      keyed by the received code.
+
+    The configuration returned (in case of success) will include:
+    * The received code;
+    * A newly generated suggested password;
+    * In case the user corresponding to the email address has verified phone
+      numbers, these will be sent (masked) to allow the user to use extra
+      security. (If the user does not use extra security, any verified NIN or
+      phone number will be unverified upon resetting the password).
+
+    As side effects, this view will:
+    * Create a MailAddressProofing element in the proofing_log;
+    * Set the email_code.is_verified flag in the PasswordResetEmailState
+      object;
+    * Set a hash of the generated password in the session.
+
+    This operation may fail due to:
+    * The code does not correspond to a valid state in the db;
+    * The code has expired;
+    * No valid user corresponds to the eppn stored in the state.
+    """
+    current_app.logger.info(f'Configuring password reset frm for {code}')
+    try:
+        state = get_pwreset_state(code)
+    except BadCode as e:
+        return error_message(e.msg)
+
+    verify_email_address(state)
+
+    new_password = generate_suggested_password()
+    salt = generate_salt()
+    hashed = b64encode(hash_password(new_password, salt)).decode('utf8')
+    session.reset_password.generated_password_hash = hashed
+    session.reset_password.generated_password_salt = salt
+
+    user = current_app.central_userdb.get_user_by_eppn(state.eppn)
+    verified_phones = user.phone_numbers.verified.to_list()
+
+    try:
+        alternatives = get_extra_security_alternatives(state.eppn)
```

Comment:
why not pass the user to the function to avoid it loading the user again from the database?

Reply:
totally right :)